test/presentation/home/widgets/password_modal_test.dart

```diff
@@ -0,0 +1,183 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:collaction_app/domain/core/i_settings_repository.dart';
+import 'package:collaction_app/presentation/home/widgets/password_modal.dart';
+import 'package:collaction_app/presentation/themes/constants.dart';
+import 'package:collaction_app/presentation/routes/app_routes.gr.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get_it/get_it.dart';
+import 'package:mocktail/mocktail.dart';
+
+import '../../../domain/core/i_settings_repository.mocks.dart';
+import '../../../test_helper.dart';
+import '../../../test_utilities.dart';
+import '../../router.mocks.dart';
+
+void main() {
+  late StackRouter stackRouter;
+  late ISettingsRepository iSettingsRepository;
+
+  setUpAll(() {
+    stackRouter = RouteHelpers.setUpRouterStubs();
+
+    iSettingsRepository = SettingsRepositoryMock();
+    when(
+      () => iSettingsRepository.addCrowdActionAccess(
+          crowdActionId: tCrowdaction.id),
+    ).thenAnswer((_) async {});
+    when(
+      () => iSettingsRepository.getCrowdActionAccessList(),
+    ).thenAnswer((_) async {
+      return [];
+    });
+    GetIt.I.registerSingleton<ISettingsRepository>(iSettingsRepository);
+  });
+
+  group('PasswordModal test:', () {
+    testWidgets('can render', (WidgetTester tester) async {
+      await buildAndPump(
+        tester: tester,
+        widget: PasswordModal(crowdAction: tCrowdaction),
+      );
+      await tester.pumpAndSettle();
+
+      expect(find.byType(PasswordModal), findsOneWidget);
+      expect(find.text('Enter password'), findsOneWidget);
+      expect(
+          find.text(
+              'This crowdaction is private. Please enter the password to see it.'),
+          findsOneWidget);
+      expect(find.byType(TextField), findsOneWidget);
+    });
+
+    testWidgets(
+      'button enabled after text input',
```

Comment:
This is just my own preference, not sure if we can adopt it to have the test names follow the `should ... when ...` format e.g `should enable submit button when text is input`

Reply:
Should we agree on a naming convention for tests going forward? I noticed there's no consistency in our test names, so I assumed it wasn't a problem, but agree that it would look nicer.